### PR TITLE
Make CustomTab Optional Dependecy

### PR DESCRIPTION
**Pull request by sojanpr**
Making Chrome custom tab optional.

@aaustin @EvangelosG @derrickstaten

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java [hunk 1]
```diff
@@ -44,9 +68,11 @@ public void checkForStrongMatch(Context context, String cookieMatchDomain, Devic
         //Check if strong match checked in last 30 days
         if (System.currentTimeMillis() - prefHelper.getLastStrongMatchTime() < THIRTY_DAYS_EPOCH_MILLI_SEC) {
             updateStrongMatchCheckFinished(callback);
+        } else if (!isCustomTabsAvailable_) {
+            updateStrongMatchCheckFinished(callback);
         } else {
             try {
-                if (deviceInfo.isHardwareIDReal() && deviceInfo.getHardwareID() != null) {
+                if (/*deviceInfo.isHardwareIDReal() && */deviceInfo.getHardwareID() != null) {
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java]: should make sure to comment this back in

File: README.md [hunk 2]
```diff
@@ -165,6 +165,14 @@ If you would like to support push notification based routing while your app alre
        intent.putExtra("branch_force_new_session",true);
 ```
 
+### Guaranteed Matching
+Branch support hundred percent guaranteed matching with cookie based matching using Custom Chrome Tabs. This is highly recommended if you like to do user authentication through deep link metadata.
+Just add the following to your build.gradle file to enable guaranteed matching
```
Comment (aaustin) [inline README.md]: Let's add a note about Multidex and how to survive it. At least, let's link to the troubleshooting guide where we have that listed.

**Comment by sojanpr** [comment]
Tested with the  following steps
1) Connect device through  proxy 
2) Build app without custom tabs dependency in live mode
3) Fresh install and Launch app. Check if app.link is called before open
4) Build app with custom tab dependency in live mode
5) Repeat 3 and  4.

**Comment by aaustin** [comment]
You checked the logs to make sure it's hitting the link service in the proper format?

**Comment by aaustin** [comment]
👍

**Pull request merged**